async_simple/util/test/ThreadPoolTest.cpp

```diff
@@ -68,4 +68,74 @@ TEST_F(ThreadPoolTest, testScheduleWithId) {
     ASSERT_TRUE(_tp->getCurrentId() == -1);
 }
 
+using namespace async_simple::util;
+
+void TestBasic(ThreadPool& pool) {
+    EXPECT_EQ(ThreadPool::ERROR_TYPE::ERROR_NONE, pool.scheduleById([] {}));
+    EXPECT_GE(pool.getItemCount(), 0);
+
+    EXPECT_EQ(ThreadPool::ERROR_TYPE::ERROR_POOL_ITEM_IS_NULL,
+              pool.scheduleById(nullptr));
+    EXPECT_EQ(pool.getCurrentId(), -1);
+
+    pool.scheduleById([&pool] { EXPECT_EQ(pool.getCurrentId(), 1); }, 1);
+}
+
+TEST(ThreadTest, BasicTest) {
+    ThreadPool pool;
+    EXPECT_EQ(ThreadPool::DEFAULT_THREAD_NUM, pool.getThreadNum());
+    ThreadPool pool1(2);
+    EXPECT_EQ(2, pool1.getThreadNum());
+
+    TestBasic(pool);
+
+    ThreadPool tp(ThreadPool::DEFAULT_THREAD_NUM, /*enableWorkSteal = */ true);
+    TestBasic(tp);
+}
+
+class ScopedTimer {
+public:
+    ScopedTimer(const char* name)
+        : m_name(name), m_beg(std::chrono::high_resolution_clock::now()) {}
+    ~ScopedTimer() {
+        auto end = std::chrono::high_resolution_clock::now();
+        auto dur =
+            std::chrono::duration_cast<std::chrono::nanoseconds>(end - m_beg);
+        std::cout << m_name << " : " << dur.count() << " ns\n";
+    }
+
+private:
+    const char* m_name;
+    std::chrono::time_point<std::chrono::high_resolution_clock> m_beg;
+};
+
+const unsigned int COUNT = 1'000'000;
+const unsigned int REPS = 10;
+
+void Benchmark(bool enableWorkSteal) {
```

Comment:
Let's add benchmarks in benchmarks directory

Reply:
ok